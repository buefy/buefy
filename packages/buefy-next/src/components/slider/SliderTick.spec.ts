import { beforeEach, describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import BSliderTick from '@components/slider/SliderTick.vue'

let wrapper: VueWrapper<InstanceType<typeof BSliderTick>>
const BSlider = {
    data() {
        return {
            _isSlider: true
        }
    },
    template: '<div><slot /></div>'
}

describe('BSliderTick', () => {
    beforeEach(() => {
        wrapper = mount(BSlider, {
            slots: {
                default: BSliderTick
            }
        }).findComponent(BSliderTick)
    })

    it('is called', () => {
        expect(wrapper.vm).toBeTruthy()
        expect(wrapper.vm.$options.name).toBe('BSliderTick')
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
