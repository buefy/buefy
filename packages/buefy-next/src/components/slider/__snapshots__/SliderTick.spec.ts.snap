// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BSliderTick > render correctly 1`] = `
"<div class=\\"b-slider-tick\\" style=\\"left: 0%;\\">
  <!--v-if-->
</div>"
`;
